polylogue/ui/__init__.py

```
@@ -52,6 +52,22 @@ def console(self) -> ConsoleLike:
     def console(self, value: ConsoleLike) -> None:
         self._facade.console = value
 
+    def progress(self, description: str, total: int):
+        """Provide a progress context manager in interactive mode; noop in plain."""
+        if self.plain:
+            return _PlainProgress()
```

Ruff (0.15.9)

[error] Undefined name `_PlainProgress`

(F821)